lib/calculator/parser.rb

```diff
@@ -63,7 +63,61 @@ def shift_tokens
 
     # Entry point for the parser, parses an <em>expression</em>
     def parse_expression
+
       # TODO implement me
+      def parse_expression
+
+        ExpressionNode.new(parse_term, parse_expression_prime)
+      end
+
+      def parse_expression_prime
+            case @tokens.first
+              when AddOpToken
+                ExpressionPrimeNode.new(parse_term, parse_expression_prime, shift_tokens)
```

Comment:
For example here, I think the shift_tokens should really be at the beginning.  If you don't consume the AddOpToken within parse_expression_prime, it will get passed down.  You probably want something more like:

ExpressionPrimeNode.new(shift_tokens, parse_term, parse_expression_prime)